specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/JavaComponents.json

```diff
@@ -360,6 +360,43 @@
     }
   },
   "definitions": {
+    "ScgRoute": {
+      "description": "Spring Cloud Gateway route definition",
+      "type": "object",
+      "required": [
+        "id",
+        "uri"
+      ],
+      "properties": {
+        "id": {
```

Comment:
Want to check, is this an azure resource id or not?

Having a writable 'id' property in an Azure rest api is unusual, it could be confusing compared to Azure resource ids.

Reply:
Would there be a good alternative name for this property?

Reply:
And maybe I'll have more questions, like what 'scope' is it uniquely identifying things in?

Reply:
No it's not a Azure resource ID. It can be values like "route1". 

The reason is that this spring cloud gateway is an open source middleware. We want to keep the naming convention with the open source configuration to let customers pick it up easier.

Reply:
This is a sample configuration for oss scg: 
![image](https://github.com/Azure/azure-rest-api-specs/assets/15007516/60e1d7d9-4267-4ecf-8724-38f343f63cff)

Reply:
OK, probably the best approach is just to be very explicit in the documentation / description of the property as to what its expected to contain / look like, and e.g. explain what SCG configuration property you're exposing.